app/src/main/java/com/epfl/beatlink/ui/components/library/PlaylistCover.kt

```diff
@@ -32,35 +38,51 @@ import com.epfl.beatlink.ui.theme.SecondaryGray
 @Composable
 fun PlaylistCover(
     coverImage: MutableState<Bitmap?>,
-    modifier: Modifier = Modifier,
+    size: Dp = 80.dp,
     isClickable: Boolean = false,
     onClick: (() -> Unit)? = null
 ) {
   // Playlist Cover
-  Box(
-      modifier =
-          Modifier.background(color = SecondaryGray, shape = RoundedCornerShape(size = 20.dp))
-              .then(
-                  if (isClickable && onClick != null) Modifier.clickable { onClick.invoke() }
-                  else Modifier)
-              .width(100.dp)
-              .height(100.dp)
-              .testTag("playlistCover"),
-      contentAlignment = Alignment.Center) {
+    GrayBox(
+        modifier = Modifier
+        .then(
+            if (isClickable && onClick != null) Modifier.clickable { onClick.invoke() }
+            else Modifier),
+        size = size
+    ) {
         coverImage.value?.let { bitmap ->
-          Image(
-              bitmap = bitmap.asImageBitmap(),
-              contentDescription = "Cover Picture",
-              modifier = modifier)
+            Image(
+                bitmap = bitmap.asImageBitmap(),
+                contentDescription = "Cover Picture",
+                contentScale = ContentScale.Crop)
         }
             ?: run {
-              // Placeholder content if no image is selected
-              Text(
-                  text = "Add \n Playlist Cover",
-                  style = MaterialTheme.typography.bodyLarge,
-                  color = PrimaryGray,
-                  textAlign = TextAlign.Center,
-                  modifier = Modifier.testTag("emptyCoverText"))
+                // Placeholder content if no image is selected
+                Text(
+                    text = "Add \n Playlist Cover",
+                    style = MaterialTheme.typography.bodyLarge,
+                    color = PrimaryGray,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier.testTag("emptyCoverText"))
```

Comment:
All this placeholder can be deleted because the case where the image does not exist is already handle.